gk/main.c

```diff
@@ -850,6 +850,94 @@ add_ggu_policy(struct ggu_policy *policy,
 	}
 }
 
+static void
+flush_flow_table(struct ip_prefix *src,
+	struct ip_prefix *dst, struct gk_instance *instance)
+{
+	uint16_t proto = 0;
+	uint32_t next = 0;
+	int32_t index;
+	uint32_t num_flushed_flows = 0;
+	const struct ip_flow *key;
+	void *data;
+	struct in_addr ip4_src_mask;
+	struct in_addr ip4_dst_mask;
+	struct in6_addr ip6_src_mask;
+	struct in6_addr ip6_dst_mask;
+
+	memset(&ip4_src_mask, 0, sizeof(ip4_src_mask));
+	memset(&ip4_dst_mask, 0, sizeof(ip4_dst_mask));
+	memset(&ip6_src_mask, 0, sizeof(ip6_src_mask));
+	memset(&ip6_dst_mask, 0, sizeof(ip6_dst_mask));
+
+	if (src->addr.proto == ETHER_TYPE_IPv4) {
+		ip4_prefix_mask(src->len, &ip4_src_mask);
+		proto = ETHER_TYPE_IPv4;
+	} else {
+		ip6_prefix_mask(src->len, &ip6_src_mask);
+		proto = ETHER_TYPE_IPv6;
+	}
+
+	if (proto == ETHER_TYPE_IPv4)
+		ip4_prefix_mask(dst->len, &ip4_dst_mask);
+	else
+		ip6_prefix_mask(dst->len, &ip6_dst_mask);
+
+	index = rte_hash_iterate(instance->ip_flow_hash_table,
+		(void *)&key, &data, &next);
+	while (index >= 0) {
+		bool matched = true;
+		struct flow_entry *fe =
+			&instance->ip_flow_entry_table[index];
+
+		if (proto != fe->flow.proto)
+			goto next;
+		else if (proto == ETHER_TYPE_IPv4) {
```

Comment:
I know that I've asked for it before, but drop the `else` and add a blank line after the `goto next;`. Sorry for the confusion.